modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/IndexRow.java

```diff
@@ -17,26 +17,26 @@
 
 package org.apache.ignite.internal.storage.index;
 
-import org.apache.ignite.internal.storage.PartitionStorage;
-import org.apache.ignite.internal.storage.SearchRow;
+import java.nio.ByteBuffer;
+import org.apache.ignite.internal.storage.MvPartitionStorage;
+import org.apache.ignite.internal.storage.RowId;
 
 /**
- * Represents an Index Row - a set of indexed columns and Primary Key columns (for key uniqueness).
+ * Represents an Index Row - a set of indexed columns and a corresponding Row ID.
  */
 public interface IndexRow {
     /**
-     * Returns the serialized presentation of this row as a byte array.
+     * Returns the serialized presentation of indexed columns as a byte array in
+     * <a href="https://cwiki.apache.org/confluence/display/IGNITE/IEP-92%3A+Binary+Tuple+Format">IEP-92</a> format.
      *
      * @return Serialized byte array value.
      */
-    byte[] rowBytes();
+    ByteBuffer indexBytes();
```

Comment:
It's weird that here we operate with bytes instead of InternalTuple or something. What's the reason of such decision? I find it non-intuitive

Reply:
Because it is impossible to deserialize an `InternalTuple`, while a `ByteBuffer` can be wrapped into a `BinaryTuple`. Returning a `BinaryTuple` might be a good alternative, but I'm not sure if it's actually better

Reply:
I think it would be better. I'd prefer InternalTuple for convenience.
Anyway, there should be a common way of converting a InternalTuple into IEP-92 data format, otherwise such API wouldn't work. I realize that, and I believe that we should provide this "common way"

Reply:
Replaced with BinaryTuple